### ISSUE TITLE
Cannot change reaction bounds

### ISSUE THREAD

**Issue by Paulocracy** [cnapy-org/CNApy#298]
Initially reported in but separated from #296:
When I change a reaction lower/upper bound, this new value is not stored for the reaction. It doesn't matter if the input is valid or not.

**Comment by axelvonkamp** [comment]
Somehow changing bounds still does not always work reliably, will investigate...

**Comment by Paulocracy** [comment]
Under which circumstances does it not work for you?

**Comment by axelvonkamp** [comment]
First of all, the validate_equation function does not appear to work for exchange reactions, probably rejects them because one side is empty.

**Comment by Paulocracy** [comment]
Yes I saw that too, it only works if there is a substrate, not if there is a product only

**Comment by axelvonkamp** [comment]
Another case: Load SmallExample and try to reduce the upper bound of R6 from 100 to 10. Although the change is valid and accepted after switching to another reaction and back to R6 the upper bound is now 1000 (the cobra default value).

**Comment by axelvonkamp** [comment]
> 
> 
> Another case: Load SmallExample and try to reduce the upper bound of R6 from 100 to 10. Although the change is valid and accepted after switching to another reaction and back to R6 the upper bound is now 1000 (the cobra default value).

The reason is that after setting the bounds in the appy function build_reaction_from_string is always called which then apparently puts in cobra default bounds. Do you get this behaviour too, Pavlos? I am pretty sure that changing bounds worked for me a copule of weeks before....

**Comment by Paulocracy** [comment]
I can't reproduce the behaviour with the SmallExample and R6, at least when I tried to change the maximum to 10 and switch between the reactions. FVAs and FBAs then also respect the new maximal bound of 10. Do I miss a reproducing step?

**Comment by axelvonkamp** [comment]
> 
> 
> I can't reproduce the behaviour with the SmallExample and R6, at least when I tried to change the maximum to 10 and switch between the reactions. FVAs and FBAs then also respect the new maximal bound of 10. Do I miss a reproducing step?

Strange, are you sure you are on the master branch?

**Comment by Paulocracy** [comment]
Yes I am :O
